scripts/common/docker.sh

```diff
@@ -207,3 +207,60 @@ function canonical_image_name() {
     fi
     echo "$image"
 }
+
+# It will only uninstall docker if is a new installation
+# and the installer has containerd set to workaround the bug issue:
+# `dpkg: no, cannot proceed with removal of containerd ... docker.io
+# depends on containerd (>= 1.2.6-0ubuntu1~)  containerd is to be removed.`
+# More info: https://bugs.launchpad.net/ubuntu/+source/docker.io/+bug/1940920
+# https://bugs.launchpad.net/ubuntu/+source/docker.io/+bug/1939140
+function uninstall_docker_new_installs_with_containerd() {
```

Comment:
this almost entirely duplicates the other function. is it possible to reuse code?

Reply:
It s not exactly the same, see that we cannot remove the same packages.